src/libecalc/core/consumers/venting_emitter.py

```diff
@@ -58,19 +51,64 @@ def evaluate(
         variables_map: VariablesMap,
     ) -> Dict[str, EmissionResult]:
         logger.debug(f"Evaluating VentingEmitter: {self.venting_emitter_dto.name}")
-        fuel_rate = self.evaluate_fuel_rate(variables_map=variables_map)
-        return self.temporal_fuel_model.evaluate_emissions(
-            variables_map=variables_map,
-            fuel_rate=np.asarray(fuel_rate),
+        emission_rate = self.evaluate_temporal(
+            variables_map=variables_map, temporal_expression=self.temporal_emission_rate_model
+        )
+        regularity = self.evaluate_temporal(
+            variables_map=variables_map, temporal_expression=self.temporal_regularity_model
         )
 
-    def evaluate_fuel_rate(self, variables_map: dto.VariablesMap) -> List[float]:
-        """We model fuel [unit-less] and emission rate [kg/day] as 1 to 1, thus we return the emission rate as
-        a proxy for fuel rate.
+        emissions = self.evaluate_venting_emissions(
+            variables_map=variables_map,
+            emission_rate=np.asarray(emission_rate),
+            emitter_model=self.venting_emitter_dto.emitter_model,
+            regularity=np.asarray(regularity),
+        )
+        return emissions
 
-        See the main docstring of this class.
-        """
+    @staticmethod
+    def evaluate_temporal(
+        variables_map: dto.VariablesMap, temporal_expression: TemporalModel[Expression]
+    ) -> List[float]:
         return TemporalExpression.evaluate(
-            temporal_expression=self.temporal_emission_rate_model,
+            temporal_expression=temporal_expression,
             variables_map=variables_map,
         )
+
+    def evaluate_venting_emissions(
+        self,
+        variables_map: dto.VariablesMap,
+        emission_rate: NDArray[np.float64],
+        regularity: NDArray[np.float64],
+        emitter_model: Dict[datetime, EmitterModel],
+    ) -> Dict[str, EmissionResult]:
+        logger.debug("Evaluating fuel usage and emissions")
+
+        # Creating a pseudo-default dict with all the emitters as keys. This is to handle changes in a temporal model.
+        emission_name = self.venting_emitter_dto.emission_name
+        emissions = {emission_name: EmissionResult.create_empty(name=emission_name, timesteps=[])}
+
+        for period, model in TemporalModel(emitter_model).items():
+            if model.convert_to_stream_day:
+                start_index, end_index = period.get_timestep_indices(variables_map.time_vector)
+                regularity_this_period = regularity[start_index:end_index]
+                emission_rate[start_index:end_index] = Rates.to_stream_day(
+                    emission_rate[start_index:end_index], regularity=list(regularity_this_period)
+                )
+
+        emission_rate_kg_per_day = emission_rate
```

Comment:
always kg? stated in docs?

Reply:
This will be solved in another PR. Allowing for input units for venting emitters. Converting to kg per day before core, converting to t/d for results.